Model/Cache/Type/Interests.php

```diff
@@ -97,4 +83,31 @@ public function productAttributeExists($attributeId) {
         return ($attr && $attr->getId());
     }
 
+    /**
+     * Reset interest product 
+     * attributes values
+     */
+    private function _resetInterestProductAttributes()
+    {
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+        
+        /** @var \Magento\Catalog\Model\ResourceModel\Product\Collection $productCollection */
+        $productCollection = $objectManager->create('Magento\Catalog\Model\ResourceModel\Product\Collection');
```

Comment:
Just a note: Let's consider injecting this class in the constuctor in the future. It's usually results in a Not Approved PR, but I know it was there before. ;)